src/Disks/ObjectStorages/IObjectStorage.h

```diff
@@ -143,28 +142,35 @@ struct RelativePathWithMetadata
     std::optional<DataFileMetaInfoPtr> file_meta_info;
     /// Retry request after short pause
     CommandInTaskResponse command;
+    std::optional<String> absolute_path;
+    std::optional<ObjectStoragePtr> object_storage_to_use = std::nullopt;
```

Comment:
Optional to pointer? That's new to me. What is wrong with just pointer = nullptr?

Reply:
oops. forgot about it. will fix